docs/index.md

```diff
@@ -1,94 +1,217 @@
 
-    Clone repo
-    ```bash
-    some bash instructions for reference
-    ```
+Direct3D11 comes with quite a bit of classes and constructs to make use of the API.
 
-=== "Ubuntu"
+- `ID3D11Device` elaborate...
 
-    long list of instructions
-    and more instruction
+- `ID3D11DeviceContext` elaborate...
 
-=== "Fedora"
+- `ID3D11Texture` elaborate...
 
-    === "SubTab1"
+- `ID3D11ShaderResourceView` elaborate...
 
-    Again instructions
-    and more stuff here
+- `ID3D11UnorderedAccessViw` elaborate...
 
-    === "SubTab2"
+- `ID3D11PixelShader` elaborate...
 
-    SubTab2Content
+- `ID3D11VertexShader` elaborate...
 
-=== "ArchLinux"
+- `ID3D11InputLayout` elaborate...
 
-    !!! info
+- `ID3D11RasterizerState` elaborate... asd
 
-        Some information inside this tab
+### Libraries & Project Layout
 
-    ```cs
-    class SomeCSharpCodeHere
+For the LearnD3D11 tutorial series we will be using the following third party libraries
+
+- [GLFW](https://www.glfw.org) It provides the window to render our things in, as well as input handling via mouse and keyboard. Yes there are also alternatives like `SDL2` for instance. It serves the same purpose, we just decided to go with `GLFW`. You might also wonder about "I want to do it myself, from scratch" then you really could do it yourself; but its the year 2022 by the time we are writing this thing and the problem of creating a window and handling input has been solved many times already, why reinvent the wheel. We have a link [here](DIYWinApi.md) to show how you could do it).
+
+- [DirectXMath](...link...) As part of the Windows SDK will be the math library we use
+
+- [Assimp](www.assimp.org) Assimp provides facilities to load mesh files of various file formats
+
+- [dear IMGUI](https://github.com/ocornut/imgui) dear IMGUI is an easy to use immediate mode graphics user interface library, its widely used in the industry and we will be using it to display interactive options
+
+#### Project Layout
+
+- `.github` contains github related files to enable this project's existence
+- `docs` contains this very documentation describing each chapter
+- `lib` contains all external/third party libraries like `GLFW` or `AssImp`
+- `src` contains all the example source code behind this tutorial series
+
+# 0. Initial Setup
+
+```todo
+either clone repo, open empty solution with project and empty main.cpp
+or explain how to create a project in VS, set it up so that it makes sense,
+explain include/lib folders, how to set output folder and why Win32 is not
+an option really anymore either
+```
+
+Assume we cloned the empty main, already setup with GLFW
+
+Ok let's start with the whole example for a HelloWindow first
+
+```cpp
+#include <GLFW/glfw3.h>
+#include <cstdint>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+
+    const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
```

Comment:
Should not be `const`

Reply:
Hmm, dumb question: why not?

Reply:
Because most GLFW functions require a non-const pointer to `GLFWmonitor`, for example `glfwGetVideoMode`

Reply:
hmm but it compiles fine with const in that constellation. And once the thing is moved into Application it will just be a constless field... Very possible i simply dont understand the thing.

Reply:
`const auto primaryMonitor = glfwGetPrimaryMonitor();` worked fine because the type of `primaryMonitor` was of `GLFWmonitor* const`, meanwhile `const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();` fails in `glfwGetVideoMode()` since its type is now `const GLFWmonitor*`.

Reply:
Ah yes, i didnt see the star on monitor :)